Automattic/wp-calypso
# Domains: inline the goToDomainCheckout function from lib/upgrades

Another divestment in `lib/upgrades/actions`. This time the `goToDomainCheckout` function that is used only at one place: after clicking on a domain suggestion in domain search UI. It decides whether the user and domain are eligible for the G Suite and either redirect to the G Suite nudge page, or directly to checkout.

This patch inlines the function's code into the only usage place, and makes use of the `isGSuiteRestricted` helper that does the check for `is_valid_google_apps_country` attribute of the user.

**How to test:**
Try to add a new domain to your site. Verify that after selecting the domain suggestion in domain search, you are taken to the page that offers G Suite if the domain is eligible.

Examples of domains that are not eligible:
- the `.in` TLD
- contains a `google` substring
